src/Server.hs

```diff
@@ -1,40 +1,54 @@
 {-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE DeriveGeneric #-}
 
-module Server (run) where
+module Server where
 
-import Control.Monad.IO.Class (liftIO)
-import Data.Text.Lazy (pack, Text, unpack)
-import Network.Wai.Middleware.RequestLogger (logStdoutDev)
-import System.Random (getStdGen, randomRIO)
+import GHC.Generics
 import Web.Scotty
+import Data.Text.Lazy (pack, Text, unpack)
+import Data.Monoid ((<>))
+import Data.Aeson (FromJSON, ToJSON)
+import Control.Monad.IO.Class (liftIO)
+import System.Random (randomRIO)
 
-run :: IO ()
-run = do
-  scotty 3000 $ do
-    middleware logStdoutDev
-    routes
+data Question = Question { questionId :: Int, questionText :: String } deriving (Show, Generic)
+instance ToJSON Question
+instance FromJSON Question
 
-routes :: ScottyM ()
-routes = do
-  get "/" $ text "hello world"
+secondManOnTheMoon :: Question
+secondManOnTheMoon = Question { questionId = 1, questionText = "Who was the second man on the moon?" }
 
-  get "/:questionId" $ do
-    questionId <- param "questionId"
-    let index = paramToInt questionId
-    let question = questions !! index
-    display ["<h1>Question ", questionId, ": ", question ,"</h1>"]
+closestPlanetToTheSun :: Question
+closestPlanetToTheSun = Question { questionId = 2, questionText = "What is the closest planet to the Sun?" }
 
-  get "/random" $ do
-    index <- liftIO $ randomRIO (0 :: Int, 1 :: Int)
-    let question = questions !! index
-    let questionId = pack $ show index
-    display ["<h1> Question ", questionId, ": ", question, "</h1>"]
+allQuestions :: [Question]
+allQuestions =  [ Question { questionId = 1, questionText = "What time is it?"}
+                , Question { questionId = 2, questionText = "Where are we?"}
+                , Question { questionId = 3, questionText = "Who are you?"}]
 
-paramToInt :: Text -> Int
-paramToInt x = read (unpack x) :: Int
+matchesId :: Int -> Question -> Bool
+matchesId id question = questionId question == id
 
-questions :: [Text]
-questions = ["Who has been nominated for the most Oscars?","What's the closest planet to the Sun?"]
+getQuestionText :: Int -> [Char]
+getQuestionText id = questionText (head (filter (matchesId id) allQuestions))
```

Comment:
And if we want to go point free
`questionText (head (filter ((id ==) . questionId) allQuestions))`